src/menu_events.cpp

```diff
@@ -591,7 +591,8 @@ bool menu_handler::end_turn(int side_num)
 void menu_handler::goto_leader(int side_num)
 {
 	unit_map::const_iterator i = units().find_leader(side_num);
-	if(i != units().end()) {
+	if(i != units().end()
+			&& i->is_visible_to_team(gui_->get_disp_context().teams()[gui_->viewing_team()], gui_->get_disp_context(), false)) {
```

Comment:
Same here.

Reply:
Also, please put the display context into a local variable to make the line shorter.

Reply:
Done. Thanks again!